logger.go

```diff
@@ -0,0 +1,234 @@
+/*
+ Copyright 2016 Derrick J. Wippler. All rights reserved.
+ Use of this source code is governed by a BSD-style
+ license that can be found in the LICENSE file.
+
+COPY NOTICE
+wrapWriter() with associated code and struct's copied from CHI middleware
+ at https://github.com/pressly/chi/blob/master/middleware/logger.go with modifications
+*/
+
+package detka
+
+import (
+	"bufio"
+	"io"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Sirupsen/logrus"
```

Comment:
The biggest draw to using logrus is the ability to include contextual information. For instance
```go
			logrus.WithFields(logrus.Fields{
				"method": "Send()",
				"type":   "mailgun",
			}).Error("sendEmail - ", err.Error())
```
If your logging system supports context information the fields can be indexed and triggers created. If not; is simple for a system like greylog to parse the bits out and make triggers for them. Structured logging is so much more powerful than flat log lines.